connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -824,10 +830,75 @@ private void transitionTo(Object connectorOrTask, TargetState state, ClassLoader
         }
     }
 
+    ConnectorStatusMetricsGroup connectorStatusMetricsGroup() {
+        return connectorStatusMetricsGroup;
+    }
+
     WorkerMetricsGroup workerMetricsGroup() {
         return workerMetricsGroup;
     }
 
+    class ConnectorStatusMetricsGroup {
+        private ConnectMetrics connectMetrics;
+        private ConnectMetricsRegistry registry;
+        private ConcurrentMap<String, MetricGroup> connectorStatusMetrics = new ConcurrentHashMap<>();
+
+
+        protected ConnectorStatusMetricsGroup(ConnectMetrics connectMetrics) {
+            this.connectMetrics = connectMetrics;
+            this.registry = connectMetrics.registry();
+        }
+
+        protected synchronized void recordTaskAdded(ConnectorTaskId connectorTaskId) {
+            if (connectorStatusMetrics.containsKey(connectorTaskId.connector())) {
+                return;
+            }
+
+            String connName = connectorTaskId.connector();
+
+            MetricGroup metricGroup = connectMetrics.group(registry.workerGroupName(),
+                registry.connectorTagName(), connName);
+
+            metricGroup.addValueMetric(registry.connectorTotalTaskCount,
+                now -> tasks.values()
+                    .stream()
+                    .filter(task -> task.id().connector().equals(connName))
+                    .count());
```

Comment:
It would be easier to test this logic if we had a method that returned the lambda function:
```
protected LiteralSupplier<Integer> taskCounter(String connName) {
  return (now) -> tasks.values()
                       .stream()
                       .filter(task -> task.id().connector().equals(connName))
                       .count());
}
```
so the highlighted lines can be:
```
            metricGroup.addValueMetric(registry.connectorTotalTaskCount, taskCounter(connName));
```